### PR TITLE
create entity AccessType

### PR DESCRIPTION
**Pull request by octap16**

### CODE REVIEW

File: src/main/java/com/softserve/actent/model/entity/AccessType.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package com.softserve.actent.model.entity;
+
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+
+import javax.persistence.*;
```
Comment (RBarnabass) [inline src/main/java/com/softserve/actent/model/entity/AccessType.java]: Set single imports

File: src/main/java/com/softserve/actent/model/entity/AccessType.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import javax.validation.constraints.NotBlank;
+
+
+@Entity
```
Comment (RBarnabass) [inline src/main/java/com/softserve/actent/model/entity/AccessType.java]: replace entity annotation closer to class declaration

File: src/main/java/com/softserve/actent/model/entity/AccessType.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+@Data
+@NoArgsConstructor
+@RequiredArgsConstructor
```
Comment (RBarnabass) [inline src/main/java/com/softserve/actent/model/entity/AccessType.java]: Remove required constructor

File: src/main/java/com/softserve/actent/model/entity/AccessType.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+@Table(name = "access_types")
+public class AccessType {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    Integer id;
```
Comment (RBarnabass) [inline src/main/java/com/softserve/actent/model/entity/AccessType.java]: field should be private

File: src/main/java/com/softserve/actent/model/entity/AccessType.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+    @NonNull
+    @NotBlank(message = "Review text can`t be empty.")
+    String type;
```
Comment (RBarnabass) [inline src/main/java/com/softserve/actent/model/entity/AccessType.java]: field should be private

**Review by Senchakmen — DISMISSED** [review]

File: src/main/java/com/softserve/actent/model/entity/AccessType.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+@Entity
+@Table(name = "access_types")
+public class AccessType {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Integer id;
+
+    @NonNull
+    @NotBlank(message = "Review text can`t be empty.")
+    private String type;
```
Comment (thebadpanda) [inline src/main/java/com/softserve/actent/model/entity/AccessType.java]: Whare is @Column ?

File: src/main/java/com/softserve/actent/model/entity/AccessType.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
+package com.softserve.actent.model.entity;
+
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+
+import javax.persistence.*;
```
Comment (md1guy) [inline src/main/java/com/softserve/actent/model/entity/AccessType.java]: single imports

**Review by Senchakmen — DISMISSED** [review]

File: src/main/java/com/softserve/actent/model/entity/AccessType.java [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+@NoArgsConstructor
+@Entity
+@Table(name = "access_types")
+public class AccessType {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Integer id;
+
+    @NonNull
+    @NotBlank(message = "Review text can`t be empty.")
```
Comment (md1guy) [inline src/main/java/com/softserve/actent/model/entity/AccessType.java]: Change message text :>

File: src/main/java/com/softserve/actent/model/entity/AccessType.java [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    @Column(unique = true, nullable = false)
+    private String type;
```
Comment (md1guy) [inline src/main/java/com/softserve/actent/model/entity/AccessType.java]: change field name to "name"

**Review by md1guy — APPROVED** [review]

**Review by onlyVone-zz — APPROVED** [review]

**Review by md1guy — APPROVED** [review]

**Review by RBarnabass — APPROVED** [review]

**Review by Senchakmen — APPROVED** [review]

**Pull request merged**